controller/src/main/java/io/pravega/controller/store/PravegaTablesScope.java

```diff
@@ -69,14 +71,8 @@ public String getName() {
         // We then retrive id from the store (in case someone concurrently created the entry or entry already existed.
         // This unique id is used to create scope specific table with unique id.
         // If scope entry exists in Scopes table, create the streamsInScope table before throwing DataExists exception
-        return Futures.handleCompose(Futures.exceptionallyComposeExpecting(storeHelper.addNewEntry(
-                SCOPES_TABLE, scopeName, newId()),
-                DATA_NOT_FOUND_PREDICATE,
-                () -> storeHelper.createTable(SCOPES_TABLE)
-                                 .thenCompose(v -> {
-                                     log.debug("table for scopes created {}", SCOPES_TABLE);
-                                     return storeHelper.addNewEntryIfAbsent(SCOPES_TABLE, scopeName, newId());
-                                 })), (r, e) -> {
+        return Futures.handleCompose(withCreateTableIfAbsent(() -> storeHelper.addNewEntry(
```

Comment:
withCreateTableIfAbsent() can also be used on Streams, ReaderGroups and KVTs

Reply:
we use it dont we?

Reply:
yes we do.